packages/ringcentral-widgets/components/AudioSettingsPanel/index.js

```diff
@@ -53,12 +54,18 @@ export default class AudioSettingsPanel extends Component {
         callVolume: newProps.callVolume,
       });
     }
-    if (newProps.inputDeviceId !== this.props.inputDeviceId) {
+    if (newProps.inputDeviceId !== this.props.inputDeviceId || all(
+      device => device.deviceId !== this.state.inputDeviceId,
+      newProps.availableInputDevices
```

Comment:
Can we validate and update inputDeviceId when availableInputDevices are changed in audio setting module?

Reply:
The inputDeviceId from AudioSetting module is correct. The code is to correct the component inputDeviceId state itself.